R/rma-tidiers.R

```diff
@@ -0,0 +1,222 @@
+#' Tidying methods for meta-analyis objects
+#'
+#' These methods tidy the results of meta-analysis objects from the metafor package
+#'
+#' @param x An `rma` created by the `metafor` package.
+#' @inheritParams tidy.lm
+#' @param include_studies Logical. Should individual studies be included in the
+#'    output?
+#' @param ... Additional arguments
+#' @param measure Measure type. See [metafor::rma()]
+#'
+#' @return A `tibble`
+#' @export
+#'
+#' @examples
+#' 
+#' library(metafor)
+#' 
+#' df <-
+#'   escalc(
+#'     measure = "RR",
+#'     ai = tpos,
+#'     bi = tneg,
+#'     ci = cpos,
+#'     di = cneg,
+#'     data = dat.bcg
+#'   )
+#' 
+#' meta_analysis <- rma(yi, vi, data = df, method = "EB")
+#'
+#' tidy(meta_analysis)
+#'
+#' @rdname tidiers
+#' 
+tidy.rma <- function(x, conf.int = FALSE, conf.level = 0.95, exponentiate = FALSE,
+                     include_studies = TRUE, measure = "GEN", ...) {
```

Comment:
Can use `measure = c("GEN", "other_option1", "other_option2")` etc to make the set of all options clear? And then use `rlang::arg_match()` to validate this argument?

Reply:
The list is very long (60!) and probably unwieldy for this? I might link to `metafor::escalc()` instead, which lists all of them. `metafor::escalc()`does check for validity internally